patches/api/0222-Add-methods-to-get-translation-keys.patch

```diff
@@ -85,6 +85,63 @@ index 4a97e73ce59c0eee77661967f1d3ac23508aae3e..d543b2b6aa131efec9b978d0b71a228f
 +        // Paper end
      }
  
+     /**
+diff --git a/src/main/java/org/bukkit/GameMode.java b/src/main/java/org/bukkit/GameMode.java
+index 938c3217f92e6d3ef9a637269c469f8359af6347..bd5328ba599a749ceb8afaab19be67035387189c 100644
+--- a/src/main/java/org/bukkit/GameMode.java
++++ b/src/main/java/org/bukkit/GameMode.java
+@@ -9,7 +9,7 @@ import org.jetbrains.annotations.Nullable;
+  * Represents the various type of game modes that {@link HumanEntity}s may
+  * have
+  */
+-public enum GameMode {
++public enum GameMode implements net.kyori.adventure.translation.Translatable { // Paper - implement Translatable
+     /**
+      * Creative mode may fly, build instantly, become invulnerable and create
+      * free items.
+@@ -35,9 +35,43 @@ public enum GameMode {
+ 
+     private final int value;
+     private static final Map<Integer, GameMode> BY_ID = Maps.newHashMap();
++    // Paper start - translation keys
++    private final String name;
++    public static final net.kyori.adventure.util.Index<String, GameMode> NAMES = net.kyori.adventure.util.Index.create(GameMode.class, mode -> mode.name); // Paper
++
++    /**
++     * {@inheritDoc}
++     * <p>
++     * This is equivalent to {@link #longTranslationKey()}
++     */
++    @Override
++    public @org.jetbrains.annotations.NotNull String translationKey() {
++        return "gameMode." + this.name;
++    }
++
++    /**
++     * Gets the longer format translation key.
++     *
++     * @return the long form key
++     * @see #translationKey()
++     */
++    public @org.jetbrains.annotations.NotNull String longTranslationKey() {
++        return this.translationKey();
++    }
++
++    /**
++     * Gets the shorter format translation key.
++     *
++     * @return the short form key
++     */
++    public @org.jetbrains.annotations.NotNull String shortTranslationKey() {
++        return "selectWorld.gameMode." + this.name;
++    }
```

Comment:
Not a fan of this method - this is using a very specific translation key (for a client-side screen).

Reply:
Why can’t it have a method? It’s a valid translation key. I’m pretty sure there are other translationckey methods that return things only used on the client.

Reply:
It's not the fact that it's something only used by the client, but the fact that it's a very specific screen's translation keys.

Reply:
I can change the method name or javadoc to mention that. But since it *is* a key based on the game mode name, I think it should be exposed.

Prob need some more people to chime in.

Reply:
I don't think it hurts to have.
However, I think that you should add an English example in the javadocs of what differentiates the longer vs shorter version.

Spectator / Spectator Mode